afs/afs-core/src/main/java/com/powsybl/afs/OrderedDependencyManager.java

```diff
@@ -0,0 +1,96 @@
+package com.powsybl.afs;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class OrderedDependencyManager {
```

Comment:
Add @author information

A general remark about this class: you have to assert that arguments are not null when it's relevant using `Object.requireNotNull`. The idea is to throw an exception as soon as possible